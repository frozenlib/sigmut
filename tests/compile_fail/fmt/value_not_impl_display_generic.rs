use sigmut::{core::Runtime, signal_format};

fn main() {
    fn f<T>(value: T) {
        let mut rt = Runtime::new();
        let _s = signal_format!("{}", value);
    }
    f(10);
}
