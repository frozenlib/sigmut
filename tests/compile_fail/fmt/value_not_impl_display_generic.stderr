error[E0277]: `T` doesn't implement `std::fmt::Display`
 --> tests/compile_fail/fmt/value_not_impl_display_generic.rs:6:18
  |
6 |         let _s = signal_format!("{}", value);
  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ `T` cannot be formatted with the default formatter
  |
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: this error originates in the macro `$crate::format_args` which comes from the expansion of the macro `signal_format` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
  |
4 |     fn f<T: std::fmt::Display>(value: T) {
  |           +++++++++++++++++++
